moped-editor/src/views/projects/projectView/ProjectNameComplex.js

```diff
@@ -0,0 +1,320 @@
+          // and alert the parent component of the change
+          props.setIsEditing(false);
+          setTimeout(() => setSnackbarState(DEFAULT_SNACKBAR_STATE), 3000);
+          props.updatedCallback();
+        });
+    }
+  };
+
+  // this is fired when the user clicks on the 'X' to cancel the edit
+  const handleCancelClick = (e) => {
+    e.preventDefault();
+    props.setIsEditing(false);
+  };
+
+  // this is fired onChange of the project name field to check if it's valid
+  const handleProjectNameChange = (e) => {
+    // Update the state variables when the input changes
+    setProjectName(e.target.value);
+    if (!e.target.value) {
+      setPrimaryTitleError(true);
+    } else {
+      setPrimaryTitleError(false);
+    }
+  };
+
+  const handleSecondaryNameChange = (e) => {
+    // Update the secondary name state variable when the input changes
+    setSecondaryName(e.target.value);
+  };
+
+  // this is fired when the snackbar's close button is clicked or by timeout
+  const handleSnackbarClose = (event, reason) => {
+    setSnackbarState(DEFAULT_SNACKBAR_STATE);
+  };
+
+  // Because the layout is non-trivially different between view and edit modes,
+  // we're going to use two different JSX blocks to render the component's content.
+
+  return (
+    <>
+      {props.isEditing ? (
+        <EditMode
+          projectName={projectName}
+          primaryTitleError={primaryTitleError}
+          handleProjectNameChange={handleProjectNameChange}
+          secondaryName={secondaryName}
+          handleSecondaryNameChange={handleSecondaryNameChange}
+          handleAcceptClick={handleAcceptClick}
+          handleCancelClick={handleCancelClick}
+          classes={classes}
+          props={props}
+        />
+      ) : (
+        <ViewMode
+          projectData={props.projectData}
+          setIsEditing={props.setIsEditing}
+          projectId={props.projectId}
+          currentPhase={props.currentPhase}
+        />
+      )}
+      <CommonJSX
+        snackbarState={snackbarState}
+        handleSnackbarClose={handleSnackbarClose}
+      />
+    </>
+  );
+};
+
+export default ProjectNameComplex;
+
+const CommonJSX = ({ snackbarState, handleSnackbarClose }) => (
+  <Snackbar
+    anchorOrigin={{ vertical: "top", horizontal: "right" }}
+    open={snackbarState.open}
+    onClose={handleSnackbarClose}
+    key={"datatable-snackbar"}
+  >
+    <Alert onClose={handleSnackbarClose} severity={snackbarState.severity}>
+      {snackbarState.message}
+    </Alert>
+  </Snackbar>
+);
+
+const EditMode = ({
+  projectName,
+  primaryTitleError,
+  handleProjectNameChange,
+  secondaryName,
+  handleSecondaryNameChange,
+  handleAcceptClick,
+  handleCancelClick,
+  classes,
+  props,
+}) => (
+  <Grid container>
+    {/* Primary project name field */}
+    <Grid item xs={12} sm={6} sx={{ paddingRight: "30px" }}>
+      <form onSubmit={(e) => handleAcceptClick(e)}>
+        <TextField
```

Comment:
@frankhereford would you add the `autoFocus` prop to to this so that we don't lose that behavior?

Reply:
100%, great catch. Thank you John. https://github.com/cityofaustin/atd-moped/pull/1302/commits/14716e22dcceb44a58e0c781fd2139f283a40e08